core/src/main/scala/org/apache/spark/rdd/UnionRDD.scala

```diff
@@ -62,7 +64,23 @@ class UnionRDD[T: ClassTag](
     var rdds: Seq[RDD[T]])
   extends RDD[T](sc, Nil) {  // Nil since we implement getDependencies
 
+  // Evaluate partitions in parallel. Partitions of each rdd will be cached by the `partitions`
+  // val in `RDD`.
+  private[spark] lazy val parallelPartitionEval: Boolean = {
```

Comment:
If I remember correctly, no one recommends reusing task VMs for MR. Too many things assumed that they weren't reused, which is a major advantage for Spark. Also, mapper VMs aren't normally calling getSplits in MR. Split planning happens on the driver/job client, which may stick around submit multiple jobs but usually not in parallel. Caching data used for split planning would typically make a difference for iterative use cases.

This commit does change the default to parallel split planning, but I'm arguing that it's a good idea to have the safety valve, not that this is a significant enough risk that we should make parallel split planning opt-in.